irods/test/helpers.py

```diff
@@ -377,3 +378,22 @@ def enableLogging(logger, handlerType, args, level_ = logging.INFO):
             logger.removeHandler(h)
 
 
+class _unlikely_value: pass
+
+@contextlib.contextmanager
+def temporarily_assign_attribute(target, attr, value, not_set_indicator = _unlikely_value()):
+    save = not_set_indicator
+    try:
+        save = getattr(target, attr, not_set_indicator)
+        setattr(target, attr, value)
+        yield
+    finally:
+        if save != not_set_indicator:
+            setattr(target, attr, save)
+        else:
+            delattr(target, attr)
+
+def server_side_sleep(session, seconds):
+    int_, frac_ = [int(_) for _ in divmod(seconds * 1.0e6 + 0.5, 1.0e6)]
```

Comment:
can you add a comment here to say what is happening and why

Reply:
Yes, will do.

Reply:
Done